source/MaterialXTest/ShaderValid.cpp

```diff
@@ -385,209 +430,324 @@ static mx::OslValidatorPtr createOSLValidator(bool& orthographicView, std::ostre
 //
 static void runGLSLValidation(const std::string& shaderName, mx::TypedElementPtr element, mx::GlslValidator& validator,
                               mx::GlslShaderGenerator& shaderGenerator, const mx::HwLightHandlerPtr lightHandler, mx::DocumentPtr doc,
-                              std::ostream& log, bool outputMtlxDoc=true, const std::string& outputPath=".")
+                              std::ostream& log, ShaderValid_TestOptions testOptions, bool outputMtlxDoc=true, const std::string& outputPath=".")
 {
-    mx::GenOptions options;
+    std::vector<mx::GenOptions> optionsList;
```

Comment:
Create a set of generation options as needed. For now it's only creating at most 2, one for complete and one for reduced. getGenerationOptions() can be modified if we want add more options later.